### PR TITLE
e2e test for memory scaler

### LINKED ISSUE #2220: Add e2e test for Memory Scaler
### Proposal

https://github.com/kedacore/keda/tree/main/tests

### Use-Case

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Ritikaa96**
_This PR add E2E tests for memory scaler_

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Fixes #2220

### CODE REVIEW

**Comment by Ritikaa96** [comment]
Hi @tomkerkhove , @JorTurFer 
PTAL.

**Review by JorTurFer — CHANGES_REQUESTED** [review]
Hi @Ritikaa96 
First, thanks a lot for this ❤️ 
There are some notation inconsistencies, I have suggested 2 changes but you should review all the places where it happens.
 
Related with the flow, I'd propose to set less resources to ensure the scaling process (1% instead of 5% for example) and also I'd propose to instead of deleting the first SO and creating another one totally different, patching current SO to update the value. You could achieve it basically creating a variable with the SO template and calling it twice with different parameters to update it. 
WDYT?

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,8 +35,9 @@
 ### Improvements
 
 - **Azure Queue:** Don't call Azure queue GetProperties API unnecessarily ([#2613](https://github.com/kedacore/keda/pull/2613))
-- **Kafka Scaler** Make "disable" a valid value for tls auth parameter ([#2608](https://github.com/kedacore/keda/issues/2608))
 - **Datadog Scaler:** Validate query to contain `{` to prevent panic on invalid query ([#2625](https://github.com/kedacore/keda/issues/2625))
+- **Kafka Scaler** Make "disable" a valid value for tls auth parameter ([#2608](https://github.com/kedacore/keda/issues/2608))
+- **Memory Scaler** Adding e2e test for the memory scaler ([#2220](https://github.com/kedacore/keda/issues/2220))
```
Comment (zroubalik) [inline CHANGELOG.md:40]: Please add this to ### Other section as it is nod directly user facing, thanks!

**Review by zroubalik — COMMENTED** [review]
Thanks for this, @JorTurFer have some great comments.

File: CHANGELOG.md (hunk 1 above)
Comment (Ritikaa96) [reply CHANGELOG.md:40]: Hi @zroubalik , do you mean only memory scaler one or both kafka and memory scaler one too?

Comment (Ritikaa96) [reply CHANGELOG.md:40]: @zroubalik Hi, I'll add the memory scaler entry to Other section.

Comment (zroubalik) [reply CHANGELOG.md:40]: yeah, just Memory scaler

Comment (Ritikaa96) [reply CHANGELOG.md]: Done in the latest update
@zroubalik

**Comment by Ritikaa96** [comment]
> Hi @Ritikaa96 First, thanks a lot for this heart There are some notation inconsistencies, I have suggested 2 changes but you should review all the places where it happens.
> 
> Related with the flow, I'd propose to set less resources to ensure the scaling process (1% instead of 5% for example) and also I'd propose to instead of deleting the first SO and creating another one totally different, patching current SO to update the value. You could achieve it basically creating a variable with the SO template and calling it twice with different parameters to update it. WDYT?

Hi @JorTurFer , Done with the changes, Thanks for the suggestions.

File: tests/scalers/memory.test.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+import * as sh from 'shelljs'
+import * as tmp from 'tmp'
+import test from 'ava'
+import { waitForDeploymentReplicaCount } from './helpers'
+
+const testNamespace = 'memory-test'
+const scaleUpValue = 1
+const scaleDownValue = 45
+const deploymentFile = tmp.fileSync()
+const ScaledObjectFile = tmp.fileSync()
```
Comment (JorTurFer) [inline tests/scalers/memory.test.ts:11]:
```suggestion
const scaledObjectFile = tmp.fileSync()
```

**Review by JorTurFer — CHANGES_REQUESTED** [review]
LGTM! ❤️ 
Only 1 small nit about one variable

**Comment by Ritikaa96** [comment]
Hi @JorTurFer , I apologize for the delay , was stuck because of an emergency. I corrected the variable PTAL.

**Review by JorTurFer — APPROVED** [review]
LGTM!
Thanks for this, let's wait till e2e passes

**Comment by JorTurFer** [comment]
Could you update the changelog @Ritikaa96 ?

**Comment by Ritikaa96** [comment]
> Could you update the changelog @Ritikaa96 ?

Sure

**Comment by Ritikaa96** [comment]
@JorTurFer PTAL

**Pull request merged**